### ISSUE TITLE
Please implement `TaskSeq.tryItem`, `tryLast`, `tryHead`, `tryPick`, `tryFind` and the non-try counterparts 

### ISSUE THREAD

**Issue by abelbraaksma** [abelbraaksma/TaskSeq#22]
Signatures should be similar like for `Seq`, with overloads for `async` versions so that the `picker` or `chooser` function can return a `Task`.

Consider:

```
TaskSeq.isEmpty
TaskSeq.head
TaskSeq.tryHead
TaskSeq.last
TaskSeq.tryLast
TaskSeq.tryExactlyOne
TaskSeq.exactlyOne
TaskSeq.item
TaskSeq.tryItem
TaskSeq.find + async version
TaskSeq.tryFind + async version
TaskSeq.pick + async version
TaskSeq.tryPick + async version
TaskSeq.filter + async version
TaskSeq.choose + async version
```

As with `Seq`, the non-try variants will raise an exception if the precondition is not met.

**Comment by abelbraaksma** [comment]
These functions have been implemented through #23. The readme has also been updated, but for completeness sake, I'll add the signatures here as well:

```
val tryHead: taskSeq: taskSeq<'T> -> Task<'T option>
val head: taskSeq: taskSeq<'T> -> Task<'T>

val tryLast: taskSeq: taskSeq<'T> -> Task<'T option>
val last: taskSeq: taskSeq<'T> -> Task<'T>  // may raise

val tryItem: index: int -> taskSeq: taskSeq<'T> -> Task<'T option>
val item: index: int -> taskSeq: taskSeq<'T> -> Task<'T>  // may raise

val tryExactlyOne: source: taskSeq<'T> -> Task<'T option>
val exactlyOne: source: taskSeq<'T> -> Task<'T>  // may raise

val choose: chooser: ('T -> 'U option) -> source: taskSeq<'T> -> taskSeq<'U>
val chooseAsync: chooser: ('T -> #Task<'U option>) -> source: taskSeq<'T> -> taskSeq<'U>
val filter: predicate: ('T -> bool) -> source: taskSeq<'T> -> taskSeq<'T>
val filterAsync: predicate: ('T -> #Task<bool>) -> source: taskSeq<'T> -> taskSeq<'T>

val tryPick: chooser: ('T -> 'U option) -> source: taskSeq<'T> -> Task<'U option>
val tryPickAsync: chooser: ('T -> #Task<'U option>) -> source: taskSeq<'T> -> Task<'U option>
val pick: chooser: ('T -> 'U option) -> source: taskSeq<'T> -> Task<'U>  // may raise
val pickAsync: chooser: ('T -> #Task<'U option>) -> source: taskSeq<'T> -> Task<'U>  // may raise

val tryFind: predicate: ('T -> bool) -> source: taskSeq<'T> -> Task<'T option>
val tryFindAsync: predicate: ('T -> #Task<bool>) -> source: taskSeq<'T> -> Task<'T option>
val find: predicate: ('T -> bool) -> source: taskSeq<'T> -> Task<'T>  // may raise
val findAsync: predicate: ('T -> #Task<bool>) -> source: taskSeq<'T> -> Task<'T>  // may raise

```

**Comment by abelbraaksma** [comment]
Prematurely closed, test CI is currently not running. Let's wait until it works again.

**Comment by abelbraaksma** [comment]
The above functions have been implemented through #23.
